lib/index.js

```diff
@@ -64,23 +64,59 @@ async function batchRequest(batchSize, pageSize, requestParams={}, method, schoo
 }
 
 /**
- * @exports
+ * @class Class to interact with the Server
  */
-module.exports = {
+
+class Banner {
+
+  /**
+    * @constructor
+    * @param {string} school The school name as given from the list of supported schools
+    * @returns {Banner} A Banner object exposing the methods available for the given school
+    */
+    constructor(school){
+      if (arguments.length < 1 || school === undefined || school === null){
+          throw new Error('Must provide school');
+      }
+
+      if (config[school] === undefined){
+          throw new Error('Unsupported school');
+      }
+      this.School = school;
+      // config[school].methods.forEach(method => {
+      //     this[methods[method].displayName] = banner[method]
+      // });
+    }
+    /**
+   *
+   * @param offset
+   * @param max
+   * @returns {Promise<any>}
+   */
```

Comment:
Why is this moved into the lib.index file? The class definition and functions are supposed to be separate. We only want to expose the methods for the given school through the constructor.